packages/sdk-middleware-http/src/http.js

```diff
@@ -146,24 +146,13 @@ export default function createHttpMiddleware({
                 return
               }
 
-              res.json().then((result: Object) => {
-                const parsedResponse: Object = {
-                  ...response,
-                  body: result,
-                  statusCode: res.status,
-                }
-
-                if (includeResponseHeaders)
-                  parsedResponse.headers = parseHeaders(res.headers)
-
-                if (includeOriginalRequest) {
-                  parsedResponse.request = {
-                    ...fetchOptions,
-                  }
-                  maskAuthData(parsedResponse.request, maskSensitiveHeaderData)
-                }
-                next(request, parsedResponse)
-              })
+              res.json()
+                .then((result: Object) => {
+                  actOnResponse(result, res)
+                })
+                .catch(()=> {
+                  actOnResponse({}, res)
```

Comment:
This could swallow invalid JSON which is not empty e.g.: `{ "foo": }`

Better would be to check if res.text is empty and then actOnResponse() else propagate error

Reply:
Updated

Reply:
👍